test/HotModuleReplacementPlugin.test.js

```diff
@@ -97,4 +97,78 @@ describe("HotModuleReplacementPlugin", () => {
 			});
 		});
 	});
+
+	it("should correct working when entry is Object and key is a number", done => {
+		const entryFile = path.join(
+			__dirname,
+			"js",
+			"HotModuleReplacementPlugin",
+			"entry.js"
+		);
+		const statsFile3 = path.join(
+			__dirname,
+			"js",
+			"HotModuleReplacementPlugin",
+			"HotModuleReplacementPlugin.test.stats3.txt"
+		);
+		const statsFile4 = path.join(
+			__dirname,
+			"js",
+			"HotModuleReplacementPlugin",
+			"HotModuleReplacementPlugin.test.stats4.txt"
+		);
+		const recordsFile = path.join(
+			__dirname,
+			"js",
+			"HotModuleReplacementPlugin",
+			"records.json"
+		);
+		try {
+			mkdirp.sync(path.join(__dirname, "js", "HotModuleReplacementPlugin"));
+		} catch (e) {
+			// empty
+		}
+		try {
+			fs.unlinkSync(recordsFile);
+		} catch (e) {
+			// empty
+		}
+		const compiler = webpack({
+			mode: "development",
+			cache: false,
+			entry: {
+				"0": entryFile
+			},
+			recordsPath: recordsFile,
+			output: {
+				path: path.join(__dirname, "js", "HotModuleReplacementPlugin")
+			},
+			plugins: [new webpack.HotModuleReplacementPlugin()],
+			optimization: {
+				namedChunks: true
+			}
+		});
+		fs.writeFileSync(entryFile, "1", "utf-8");
+		compiler.run((err, stats) => {
+			if (err) throw err;
+			const jsonStats = stats.toJson();
+			const hash = jsonStats.hash;
+			const trunkName = Object.keys(jsonStats.assetsByChunkName)[0];
+			fs.writeFileSync(statsFile3, stats.toString());
+			compiler.run((err, stats) => {
+				if (err) throw err;
+				fs.writeFileSync(statsFile4, stats.toString());
+				fs.writeFileSync(entryFile, "2", "utf-8");
+				compiler.run((err, stats) => {
+					if (err) throw err;
+					fs.writeFileSync(statsFile3, stats.toString());
+					const result = JSON.parse(
+						stats.compilation.assets[`${hash}.hot-update.json`].source()
+					)["c"][`${trunkName}`];
+					expect(result).toBe(true);
+					done();
+				});
+			});
+		});
+	});
```

Comment:
This test case doesn't really run the code you changed, does it?

You can take a look at the `test/hotCases` folder where the tests for HMR are placed. You can `module.hot.accept()` the entry point.

Reply:
@sokra Hi, thanks for your review, but i think, this case cover my changed, because when i run the case in the older code, i got an error.  i look at the `test/hotCases` folder, but i dont know  how can i run these cases